### DISCUSSION TITLE
ServiceValueResolver extremely slow

### DISCUSSION THREAD

**Discussion by kerwanp** [symfony/symfony#54164] [Q&A]
I started investigating some performance issues we have with our large Symfony 6.2 codebase on a specific part of our project (user registration). 

As shown on this profiling graph (locally) we have two really slow part and after optimizing them and reach 700ms.

![image](https://github.com/symfony/symfony/assets/36955373/17859da7-bcd0-477c-bf8a-afe4e0f04134)

A deploy later on our staging server we still had ~1.8s of total execution time. After setting up the WebProfiler on the environment we realized that we have a huge performance issue on the `ServiceValueResolver::resolve` which take over a second.

![image (1)](https://github.com/symfony/symfony/assets/36955373/866099ca-10a9-4518-bb67-fb8a5f8b266e)

It seems to be related to the dependency injection taking a huge amount of time, but even after following the [best practices to improve performances](https://symfony.com/doc/current/performance.html#performance-configure-opcache), the numbers are the same.

> The difference between execution time of `ServiceValueResolver::resolve` on first and second graph is surely due to the hardware (local vs staging). 

What could be the culprit? Too much services? Using YML instead of attributes for routes?

**Comment by MatTheCat** [comment]
> What could be the culprit? Too much services? Using YML instead of attributes for routes?

Those shouldn’t matter once the cache is built :thinking: 

Would you have a blackfire profile in order to pinpoint the issue?
